portfolio/src/main/webapp/script.js

```diff
@@ -83,26 +83,30 @@ function deleteComment(comment) {
 
 
 function UserLogin(){
-    getEmail();
-    fetch('/user?loggedIn').then(loggedIn => loggedIn.text()).then(loggedIn => String(loggedIn)).then(loggedIn => loggedIn.trim()).then((loggedIn) => {
-        if(loggedIn[0] == 'N'){
+    fetch('/user?Logs').then(response => response.json()).then((Logs) => {
+        loggedInStatus = Logs[0]; // since we have an array
+        console.log(loggedInStatus["status"]);
+        console.log(loggedInStatus["loginUrl"]);
+        console.log(loggedInStatus["logoutUrl"]);
+
+        if(loggedInStatus["status"] == "Out"){
           document.getElementById("userOnly").style.display = "none";
           document.getElementById("notLoggedIn").style.display = "block";
           document.getElementById("logOut").style.display = "none";
+          document.getElementById("logInButton").href = loggedInStatus["logoutUrl"];//shouldn't this redirect to login?
```

Comment:
for both the login and logout buttons, they no longer take me to that login page where I can enter my email and hit login. I think this is because I deleted that from the server but I'm not quite sure how to rewrite it in a way that works with my Json format now. What do you reccomend?

Reply:
Your JS and Java looks correct.  Problem is I don't think `<button>` has an 'href` attribute.  [see docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button).

If you want to keep it as a button, you'll need to add an `onclick="JAVASCRIPT_TO_REDIRECT_HERE"`.  Another option (which I prefer) would be to switch to `<a>` and just use href